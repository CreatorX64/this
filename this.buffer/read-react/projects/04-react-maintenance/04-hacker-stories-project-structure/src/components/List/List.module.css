/* If using CSS-in-JS, where no CSS file is needed, one could still have a
separate style.js file for all the styled components */

.item {
  display: flex;
  align-items: center;
  padding-bottom: 5px;
}

.item > span {
  overflow: hidden;
  padding: 0 5px;
  white-space: nowrap;
  text-overflow: ellipsis;
}

.item > span > a {
  color: inherit;
}
