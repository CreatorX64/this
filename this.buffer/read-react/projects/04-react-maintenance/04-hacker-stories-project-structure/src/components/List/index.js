import styles from "./List.module.css";

const List = ({ list, onRemoveItem }) => (
  <ul>
    {list.map((item) => (
      <Item key={item.objectID} item={item} onRemoveItem={onRemoveItem} />
    ))}
  </ul>
);

export const Item = ({ item, onRemoveItem }) => (
  <li className={styles.item}>
    <span style={{ width: "40%" }}>
      <a href={item.url}>{item.title}</a>
    </span>
    <span style={{ width: "30%" }}>{item.author}</span>
    <span style={{ width: "10%" }}>{item.num_comments}</span>
    <span style={{ width: "10%" }}>{item.points}</span>
    <span style={{ width: "10%" }}>
      <button
        type="button"
        className="button button-small"
        onClick={() => onRemoveItem(item)}
      >
        Dismiss
      </button>
    </span>
  </li>
);

export default List;
