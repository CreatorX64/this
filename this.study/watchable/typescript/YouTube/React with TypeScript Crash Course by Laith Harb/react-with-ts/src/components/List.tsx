import { FC } from "react";
import { Person } from "../models/Person";

interface IProps {
  people: Person[];
}

export const List: FC<IProps> = ({ people }) => {
  const renderList = (): JSX.Element[] => {
    return people.map((person) => (
      <li key={person.name} className="List">
        <div className="List-header">
          <img
            className="List-img"
            src={person.url}
            alt={`Portrait shot of ${person.name}`}
          />
          <h2>{person.name}</h2>
        </div>
        <p>{person.age} years old</p>
        <p className="List-note">{person.note}</p>
      </li>
    ));
  };

  return <ul>{renderList()}</ul>;
};
