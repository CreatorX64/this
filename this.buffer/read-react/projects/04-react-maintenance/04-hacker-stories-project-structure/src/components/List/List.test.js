import { render, screen, fireEvent } from "@testing-library/react";
import { storyOne, stories } from "../../dummy-data";
import List, { Item } from "./index";

describe("List", () => {
  test("renders snapshot", () => {
    const { container } = render(<List list={stories} />);
    expect(container).toMatchSnapshot();
  });
});

describe("Item", () => {
  test("renders all properties", () => {
    render(<Item item={storyOne} />);

    // The debug() function logs the rendered component to the console. You
    // should form the habit of using RTL’s debug function whenever you
    // render a new component in a React component test. The function gives
    // a useful overview of what is rendered and informs the best way to
    // proceed with testing
    // screen.debug();

    expect(screen.getByText("Jordan Walke")).toBeInTheDocument();
    expect(screen.getByText("React")).toHaveAttribute(
      "href",
      "https://reactjs.org"
    );
  });

  test("renders a clickable dismiss button", () => {
    render(<Item item={storyOne} />);

    // The getByRole function is usually used to retrieve elements by aria-label
    // attributes. However, there are also implicit roles on HTML elements – like
    // button for a button element. Thus you can select elements not only by visible
    // text, but also by their (implicit) accessibility role with React Testing
    // Library. A neat feature of getRoleBy is that it suggests roles if you provide
    // a role that’s not available. Both, getByText and getByRole are RTL’s most
    // widely used search functions
    expect(screen.getByRole("button")).toBeInTheDocument();
  });

  test("clicking the dismiss button calls the callback handler", () => {
    // Create mock/spy/stub function
    const handleRemoveItem = jest.fn();

    // Provide the mock function as a callback handler to Item component
    render(<Item item={storyOne} onRemoveItem={handleRemoveItem} />);

    // Fire click event on the button
    fireEvent.click(screen.getByRole("button"));

    expect(handleRemoveItem).toHaveBeenCalledTimes(1);
  });

  test("renders snapshot", () => {
    const { container } = render(<Item item={storyOne} />);
    expect(container).toMatchSnapshot();
  });
});
