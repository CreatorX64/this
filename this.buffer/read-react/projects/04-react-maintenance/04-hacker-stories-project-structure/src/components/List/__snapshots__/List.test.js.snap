// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item renders snapshot 1`] = `
<div>
  <li
    class="item"
  >
    <span
      style="width: 40%;"
    >
      <a
        href="https://reactjs.org"
      >
        React
      </a>
    </span>
    <span
      style="width: 30%;"
    >
      Jordan Walke
    </span>
    <span
      style="width: 10%;"
    >
      3
    </span>
    <span
      style="width: 10%;"
    >
      4
    </span>
    <span
      style="width: 10%;"
    >
      <button
        class="button button-small"
        type="button"
      >
        Dismiss
      </button>
    </span>
  </li>
</div>
`;

exports[`List renders snapshot 1`] = `
<div>
  <ul>
    <li
      class="item"
    >
      <span
        style="width: 40%;"
      >
        <a
          href="https://reactjs.org"
        >
          React
        </a>
      </span>
      <span
        style="width: 30%;"
      >
        Jordan Walke
      </span>
      <span
        style="width: 10%;"
      >
        3
      </span>
      <span
        style="width: 10%;"
      >
        4
      </span>
      <span
        style="width: 10%;"
      >
        <button
          class="button button-small"
          type="button"
        >
          Dismiss
        </button>
      </span>
    </li>
    <li
      class="item"
    >
      <span
        style="width: 40%;"
      >
        <a
          href="https://redux.js.org"
        >
          Redux
        </a>
      </span>
      <span
        style="width: 30%;"
      >
        Dan Abramov, Andrew Clark
      </span>
      <span
        style="width: 10%;"
      >
        2
      </span>
      <span
        style="width: 10%;"
      >
        5
      </span>
      <span
        style="width: 10%;"
      >
        <button
          class="button button-small"
          type="button"
        >
          Dismiss
        </button>
      </span>
    </li>
  </ul>
</div>
`;
